### PR TITLE
use std::shared_ptr instead of pointer of instance

### PR DESCRIPTION
**Pull request by jijoongmoon**
using std::shared_ptr to handle heap memory space

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: api/capi/src/nntrainer.cpp [hunk 1]
```diff
@@ -41,7 +41,8 @@ static int nn_object(ml_nnmodel_h *model) {
   *model = nnmodel;
 
   try {
-    nnmodel->network = new nntrainer::NeuralNetwork();
+    nnmodel->network =
+      std::shared_ptr<nntrainer::NeuralNetwork>(new nntrainer::NeuralNetwork());
```
Comment (myungjoo) [inline api/capi/src/nntrainer.cpp:45]: ```nnmodel->network = std::make_shared<nntrainer::NeuralNetwork>();``` ?

File: api/capi/src/nntrainer.cpp [hunk 2]
```diff
@@ -110,10 +111,10 @@ int ml_nnmodel_destruct(ml_nnmodel_h model) {
 
   ML_NNTRAINER_CHECK_MODEL_VALIDATION(nnmodel, model);
 
-  nntrainer::NeuralNetwork *NN;
+  std::shared_ptr<nntrainer::NeuralNetwork> NN;
   NN = nnmodel->network;
   NN->finalize();
-  delete NN;
+  // delete [] NN;
```
Comment (myungjoo) [inline api/capi/src/nntrainer.cpp:117]: Please remove the whole line, not commenting out it.

Comment (jijoongmoon) [reply api/capi/src/nntrainer.cpp]: yes. fiexd. thanks.

File: api/capi/src/nntrainer.cpp (hunk 1 above)
Comment (jijoongmoon) [reply api/capi/src/nntrainer.cpp]: yes. looks better. fixed.

**Pull request merged**